src/dotnet/APIView/APIViewWeb/Managers/CommentsManager.cs

```diff
@@ -148,6 +165,98 @@ public async Task<CommentItemModel> UpdateCommentAsync(ClaimsPrincipal user, str
+                return;
+            }
+
+            string response;
+            try
+            {
+                /*
+                var copilotEndpoint = _configuration["CopilotServiceEndpoint"];
+                var getCommentReplyEndpoint = $"{copilotEndpoint}/agent/chat";
+
+
+                IEnumerable<CommentItemModel> threadComments =
+                    await _commentsRepository.GetCommentsAsync(reviewId: comment.ReviewId, lineId: comment.ElementId);
+
+                var activeApiRevision = await _apiRevisionsManager.GetAPIRevisionAsync(apiRevisionId: activeRevisionId);
+                var activeCodeFile = await _codeFileRepository.GetCodeFileAsync(activeApiRevision, false);
+                var activeCodeLines = activeCodeFile.CodeFile.GetApiLines(skipDocs: true);
+
+                Dictionary<string, int> lineIdToIndex = activeCodeLines
+                    .Select((line, id) => new { line.lineId, id })
+                    .ToDictionary(x => x.lineId, x => x.id);
+
+                List<CommentModelForCopilot> commentsForAgent = threadComments
+                    .Select(threadComment => new CommentModelForCopilot
+                    {
+                        Author = threadComment.CreatedBy,
+                        CommentText = threadComment.CommentText,
+                        Timestamp = threadComment.CreatedOn,
+                        LineNumber = lineIdToIndex.TryGetValue(threadComment.ElementId, out var id) ? id : -1
+                    })
+                    .ToList();
+
+                var client = _httpClientFactory.CreateClient();
+                var payload = new Dictionary<string, object> { { "comments", commentsForAgent } };
+                response = await client.PostAsync(getCommentReplyEndpoint, new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json"));
+                response.EnsureSuccessStatusCode();
+                */
+
+                response = "Agent integration is not currently available.";
+            }
+            catch
+            {
+                response = "An error occurred while attempting to contact the agent service.";
+            }
+
+            commentResult.CommentText = response;
+            await _commentsRepository.UpsertCommentAsync(commentResult);
+        }
+
+        private bool IsUserAllowedToChatWithAgent(ClaimsPrincipal name, string language)
+        {
+            if (name == null || string.IsNullOrEmpty(language))
+            {
+                return false;
+            }
+
+            return IsUserLanguageArchitect(name, language);
+        }
+
+        private bool IsUserLanguageArchitect(ClaimsPrincipal user, string language)
+        {
+            string githubUser = user.GetGitHubLogin();
+
+            if (string.IsNullOrEmpty(githubUser) || string.IsNullOrEmpty(language))
+                return false;
+
+            string architects = _configuration[$"Architects:{language}"] ?? "";
```

Comment:
Are we changing the configuration now to put separate config for ech language?

Reply:
You can use the following endpoint in the APIView API to get this info:
GET `/api/Reviews/allowedApprovers`

Reply:
that is not language specific, is that ok?

Reply:
This? The reviewId would basically encode the language.
GET `/api/Reviews/{reviewId}/preferredApprovers`

Reply:
I don't think that is the endpoint, I was looking at the code and what it does is 
- get _configuration["approvers"]; (all the approvers)
- filter them base on your profile (do you remember last release that you were not able to see the button because you had specific some filter in here? it is this code)
![image](https://github.com/user-attachments/assets/793648c9-cb10-4ef6-9441-fcb45b3eeae6)


@chidozieononiwu is there another endpoint? :)